src/main/java/org/openstreetmap/atlas/checks/database/FlagDatabaseSubCommand.java

```diff
@@ -0,0 +1,375 @@
+     * Add CheckFlag values to parameterized sql INSERT statement
+     *
+     * @param sql
+     *            - PreparedStatement to add parameterized values to
+     * @param flag
+     *            - CheckFlag to insert into flag table
+     */
+    private void batchFlagStatement(final PreparedStatement sql, final CheckFlag flag)
+    {
+        try
+        {
+            sql.setString(1, flag.getIdentifier());
+            sql.setString(2, flag.getChallengeName().orElse(""));
+            sql.setString(THREE, flag.getInstructions().replace("\n", " ").replace("'", "''"));
+            sql.setObject(FOUR, this.timestamp);
+
+            sql.addBatch();
+        }
+        catch (final SQLException error)
+        {
+            logger.error("Unable to create Flag {} SQL statement", flag.getIdentifier());
+        }
+    }
+
+    /**
+     * Fet all geojson features which do contain a properties field from a {@link JsonArray}.
+     *
+     * @param features
+     *            a {@link JsonArray} of geojson features
+     * @return a JsonArray containing all Check flag features
+     */
+    private JsonArray filterOutPointsFromGeojson(final JsonArray features)
+    {
+        return StreamSupport.stream(features.spliterator(), false).map(JsonElement::getAsJsonObject)
+                .filter(feature -> feature.has(PROPERTIES)
+                        && !feature.get(PROPERTIES).getAsJsonObject().entrySet().isEmpty())
+                .collect(JsonArray::new, JsonArray::add, JsonArray::addAll);
+    }
+
+    /**
+     * Determine whether or not this file is something we can handle, and classify it accordingly.
+     *
+     * @param logFile
+     *            any file
+     * @return if this file is something this command can handle, the appropriate OutputFileType
+     *         enum value; otherwise, an empty optional.
+     */
+    private Optional<OutputFileType> getOptionalOutputType(final File logFile)
+    {
+        // Note that technically the true extension is just .gz, so we can't use the same method as
+        // below.
+        if (logFile.getName().endsWith(ZIPPED_LOG_EXTENSION))
+        {
+            return Optional.of(OutputFileType.COMPRESSED_LOG);
+        }
+        else if (FilenameUtils.getExtension(logFile.getName()).equals(LOG_EXTENSION))
+        {
+            return Optional.of(OutputFileType.LOG);
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Returns the OSM identifier for a given JsonObject. Atlas Checks OSM identifier changed from
+     * "osmid" to "osmIdentifier"
+     * {@link <a href="https://github.com/osmlab/atlas-checks/pull/116/files">here</a>}
+     *
+     * @param properties
+     *            CheckFlag properties
+     * @return OSM identifier
+     */
+    private long getOsmIdentifier(final JsonObject properties)
+    {
+        return properties.get("osmid") == null ? properties.get("osmIdentifier").getAsLong()
+                : properties.get("osm_id").getAsLong();
```

Comment:
Should this be 
```suggestion
                : properties.get("osmid").getAsLong();
```
?

Reply:
yes! also, the commit suggestion is awesome, I'm about to try it